### ISSUE TITLE
Porting to Mu4

### ISSUE THREAD

**Issue by Jojo-Schmitz** [BernardGreenberg/MuseScorePlugins#11]
Here's a port of one of these plugin to Mu4: [articulate.zip](https://github.com/BernardGreenberg/MuseScorePlugins/files/11870430/articulate.zip)

**Comment by BernardGreenberg** [comment]
This is great.  When I test it in MS3, I will post it.  Will see about the others...Thank you!

**Comment by Jojo-Schmitz** [comment]
IIRC dockable plugins are not (yet) possible in Mu4

**Comment by XiaoMigros** [comment]
I remember hearing from Marc Sabatella that even with the 'typical' Mu4 edits the plugins no longer function, or at least, they don't affect how playback sounds

**Comment by BernardGreenberg** [comment]
I have downloaded and installed MS4.1, ran it, and placed your articulate.qml in its Plugins directory, but I can't find the way to make it aware of the plugin. ReloadPlugins doesn't do it.  What do I have to do?


Thanks....
________________________________

**Comment by XiaoMigros** [comment]
'Reload plugins' doesn't do anything at the moment, have you tried restarting the program?

**Comment by BernardGreenberg** [comment]
Just tried that.  Certainly works (plugin appears where it should), but clicking on it in the plugins menu (after selecting a measure or note) crashes the app immediately.

Guess it's not ready yet.
________________________________

**Comment by XiaoMigros** [comment]
I can't immediately see a reason why, but I think it's worth noting that this plugin would not be functional (the play events system isnt present in MuseScore 4 yet)

**Comment by BernardGreenberg** [comment]
Uh, this is the version Jojo posted to this thread because he said that it IS functional!  Do read it!  Can you and he come up with one story?

Do you duplicate my result, or his?  The only reason I downloaded MS4 was to check this claim.

If the Play Event system really isn't present in MS4, I don't see how the claim that this worked could possibly be true.
________________________________

**Comment by Jojo-Schmitz** [comment]
I'm currently and for the next 2 weeks away for m.my computer...

**Comment by XiaoMigros** [comment]
For me, it's not even showing up at all in the plugin manager tab

**Comment by XiaoMigros** [comment]
![image](https://github.com/BernardGreenberg/MuseScorePlugins/assets/116587995/6ec83b1e-98f8-4308-9c10-32ea99659528)
Here are marc's notes on the play events system, for what it's worth (I haven't tested anything myself)

**Comment by BernardGreenberg** [comment]
Great.  I'll uninstall MS4. Not ready yet.  Don't know what Jojo was thinking.,
________________________________

**Comment by Jojo-Schmitz** [comment]
> For me, it's not even showing up at all in the plugin manager tab

It does for me:
![image](https://github.com/BernardGreenberg/MuseScorePlugins/assets/1786669/fcc31658-b426-4ffc-b221-4f4437bcbf66)
![image](https://github.com/BernardGreenberg/MuseScorePlugins/assets/1786669/3cc18fed-6533-41d4-897c-6e92b9612976)
![image](https://github.com/BernardGreenberg/MuseScorePlugins/assets/1786669/f9e112cc-611e-4ced-8203-4cc2a50d26e2)

**Comment by XiaoMigros** [comment]
Interesting it shows up for you, it still doesn't for me..

Regardless, as Marc said,even though the properties exist and there are no errors encountered when writing them, no changes are made to playback

**Comment by BernardGreenberg** [comment]
It showed up for me after I restarted the app.  But, as I said (this is in 4.1), as soon as I tried to apply it to notes, the app crashed, i.e.. it did not show the dialog.

Does it indeed actually work (change in notes and sound), or not (for you)? Have you changed it since the earlier version you posted?
________________________________

**Comment by Jojo-Schmitz** [comment]
I don't think it actually changed the sound, but it works and apparently changes the score in some form or shape. Haven't investigated in to what the changes are exactly. And no, I've not modified the plugin further, it is the one from the initial post.

**Comment by BernardGreenberg** [comment]
Then I cannot understand why it sort-of-works for you and crashes for me. The 4.1 I installed was out of the box and not customized in any way. I do not have MuseSounds.  Obviously, I can't post it if it doesn't work for me.
________________________________

**Comment by Jojo-Schmitz** [comment]
Anything helpful in the logs?

**Comment by BernardGreenberg** [comment]
Where do I find them? (I am running the Mac build, which may have something to do with the failures, too. What OS are you using?)
________________________________

**Comment by Jojo-Schmitz** [comment]
I'm on Windows. The logs should be in `~/Library/Application\ Support/MuseScore/MuseScore4/logs` on a Mac

**Comment by BernardGreenberg** [comment]
Thanks, will try.
________________________________

**Comment by BernardGreenberg** [comment]
Using 4.1.1, just installed, I selected a note, tried to invoke the plugin from the men and the app crashed immediately.  The log is enclosed and tells me nothing, given tht it worked for you.

I feel no urge to debug this given that the app is buggy and I don't need MS4.
________________________________

**Comment by Jojo-Schmitz** [comment]
There no log...

BTW: the plugin uitseld really works, it adds (something like) this to the score (to a `note` actually:
```
              <Events>
                <Event>
                  <len>111</len>
                  </Event>
                </Events>
 ```
It is just nothing there in Mu4 to interprete that, I guess

**Comment by Jojo-Schmitz** [comment]
Probably indeed someone needs to look at it in a Debugger
